src/unix/fs.c

```diff
@@ -129,8 +129,27 @@
 static ssize_t uv__fs_fdatasync(uv_fs_t* req) {
 #if defined(__linux__) || defined(__sun) || defined(__NetBSD__)
   return fdatasync(req->file);
-#elif defined(__APPLE__) && defined(SYS_fdatasync)
-  return syscall(SYS_fdatasync, req->file);
+#elif defined(__APPLE__) && defined(F_FULLFSYNC)
```

Comment:
I think we did that for compatibility with OS X 10.4 or 10.5 but we no longer support those.  You can remove them if you like.